### PR TITLE
Go advanced

### PR DESCRIPTION
**Pull request by shubhadapaithankar**
This PR divided in to two parts

**1. Concurrency**
- This PR is related to the website checking quicker by allowing it to check multiple websites at the same time.
- Added to tests to make sure everything works as it should and to see how much faster it is.

Testing 
```
Concurrency % go test -bench=.
goos: darwin
goarch: arm64
pkg: test/hello/Concurrency
BenchmarkCheckWebsites-10             55          21021670 ns/op
PASS
ok      test/hello/Concurrency  1.500s
```
**2. Select**

-  This PR introduces the WebsiteRacer function, which takes two URLs and "races" them by making HTTP GET requests to each.

### CODE REVIEW

File: Concurrency/check_website_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+		"http://google.com",
+		"https://www.python.org",
+		"http://test.com",
+	}
+
+	want := map[string]bool{
+		"http://google.com":      true,
+		"https://www.python.org": true,
+		"http://test.com":        false,
+	}
```
Comment (SudoBrendan) [inline Concurrency/check_website_test.go:29]: I love the idea of having a very straightforward `want` map here. However, by also defining `websites` we create a way for future changes to lead to strange results (e.g. what happens when you add a URL to `websites` but not `want`?). It might be better to stick with only defining our list in one place (probably `want`), and then constructing our URL list from that, ah lah https://stackoverflow.com/questions/21362950/getting-a-slice-of-keys-from-a-map

**Review by SudoBrendan — APPROVED** [review]

File: test
Comment (SudoBrendan) [inline test:1]: I'd remove the file from the PR to clean it up.

File: Select/racer.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package racer
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// Compares the response times of a and b, returning the fastest one.
```
Comment (SudoBrendan) [inline Select/racer.go]: nit: I'd reword this slightly to say `fetching urls a and b` so at a high level we understand what `a` and `b` are supposed to be from the start.

File: Select/racer_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		defer fastServer.Close()
+
+		slowURL := slowServer.URL
+		fastURL := fastServer.URL
+
+		want := fastURL
+		got, err := Racer(slowURL, fastURL, 10*time.Second)
+
+		if err != nil {
+			t.Fatalf("did not expect an error but got one %v", err)
```
Comment (SudoBrendan) [inline Select/racer_test.go:26]: it's better to use `%w` compared to `%v` for type Error. See https://stackoverflow.com/questions/61283248/format-errors-in-go-s-v-or-w

File: Select/racer_test.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		}
+
+		if got != want {
+			t.Errorf("got %q, want %q", got, want)
+		}
+	})
+
+	t.Run("returns an error if a server doesn't respond within 10s", func(t *testing.T) {
```
Comment (SudoBrendan) [inline Select/racer_test.go]: this test description specifies `10s` when the specified timeout is `20ms`. To fix this and make it easier for changes, I'd just reference "if both servers don't respond by the timeout"

File: Concurrency/check_website_test.go (hunk 1 above)
Comment (shubhadapaithankar) [reply Concurrency/check_website_test.go:29]: I agree .It's a good idea to define our list in one place, like 'want', to avoid future inconsistencies.

File: Select/racer_test.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		server := makeDelayedServer(25 * time.Millisecond)
+
+		defer server.Close()
+
+		_, err := Racer(server.URL, server.URL, 20*time.Millisecond)
```
Comment (SudoBrendan) [inline Select/racer_test.go:39]: nit: small latencies can eventually really add up over time. I'd suggest if we are testing the timeout functionality, we set it to something like `1ms` (or even shorter if possible).

File: test
Comment (shubhadapaithankar) [reply test:1]: Okay, sure

File: Select/racer.go (hunk 2 above)
Comment (shubhadapaithankar) [reply Select/racer.go]: ack

File: Select/racer_test.go (hunk 3 above)
Comment (shubhadapaithankar) [reply Select/racer_test.go:26]: ack

File: Select/racer_test.go (hunk 5 above)
Comment (shubhadapaithankar) [reply Select/racer_test.go:39]: ack
